tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/matter/matterusers.controller.js

```diff
@@ -1147,6 +1231,47 @@
             }
         }
 
+
+        cm.updateSuccessCallList = 0;
+        function updatePermissionsOnMatterLists(updateMatterDetails, listNumbers,listCount) {
+             if (cm.updateSuccessCallList == 0) {
+                 cm.successMsg = "Updating the user permissions to the " + cm.matterName + " list";
+            }
+            else if (cm.updateSuccessCallList == 1) {
+                cm.successMsg = "Updating the user permissions to the " + cm.matterName + " OneNoteLibrary";
+            }
+            else if (cm.updateSuccessCallList == 2) {
+                cm.successMsg = "Updating the user permissions to the " + cm.matterName + " Calendar";
+            }
+            else if (cm.updateSuccessCallList == 3) {
+                cm.successMsg = "Updating the user permissions to the " + cm.matterName + "Task";
+            }            
+            else if (cm.updateSuccessCallList == 4) {
+                cm.successMsg = "Updating the  "+ cm.matterName + " metadata.";
+            }
+            updateMatterDetails.MethodNumber = parseInt(listNumbers[cm.updateSuccessCallList]);
+            updateMatterPermissions(updateMatterDetails, function (response) {
+                if (!response.isError) {
+                    cm.updateSuccessCallList++;                   
+                    if (cm.updateSuccessCallList == listCount) {                       
+                        cm.successMsg = "Successfully updated the  permissions to the "+cm.matterName;                        

```

Comment:
@venkatesh127 move this to out of updatematterpermissions method so it will become async and improves performance.